Fix: Finding hosted zone id with matching root and sub hosted zones

Spaideri opened the pull request:
There was an issue that function getHostedZoneId() would return incorrect hosted zone id if hosted zones contained two hosted zones with matching ending for example: 
sub.root.com
root.com

When using domain name test.sub.root.com getHostedZoneId returned id for root.com instead of sub.root.com because the find was just looking to match the ending.

Reproduced the scenario on the unit test and fixed the issue. Sorting of the hosted zones became obsolete after the fix.

aoskotsky-amplify commented on index.js:
```diff
@@ -279,18 +264,19 @@ class ServerlessCustomDomain {
+        const targetHostedZone = data.HostedZones
+          .filter((hostedZone) => {
+            const hostedZoneName = hostedZone.Name.endsWith('.') ? hostedZone.Name.slice(0, -1) : hostedZone.Name;
+            return (this.targetHostedZone || '').endsWith(hostedZoneName);
```
Its a bit confusing that we have 2 variables with the same name `this.targetHostedZone` and the local variable `targetHostedZone`. Maybe one of those should be renamed. Otherwise this looks good

aoskotsky-amplify replied:
Also when would `this.targetHostedZone` ever be missing?

aoskotsky-amplify commented:
Thanks for this PR. Overall looks good except for a small nitpick. Let me know what you think and we can merge it.

Spaideri commented:
Thanks for the feedback. Renamed the overlapping variable name. I preserved the null check for the this.targetHostedZone from the previous implementation because removing it broke the 'Fail getHostedZone' unit test. Figured out the test was most likely falsely initialised, so fixed the test and removed the null check.

aoskotsky-amplify commented:
Nice. Thanks for the fix
